### PR TITLE
Bug 1245815 - Document parts of the release process and improve some of the scripts/processes

### PR DESCRIPTION
**Pull request by Standard8**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -357,6 +359,9 @@ dist_export:
 	@mkdir -p $(DIST_EXPORT_DIR)/test/functional
 	$(RSYNC) test/functional/* $(DIST_EXPORT_DIR)/test/functional
 
+.PHONY: locale_update
+locale_update: $(VENV)
+	$(VENV)/bin/python bin/locale_update.py
```
Comment (Mardak) [inline Makefile:364]: Not sure if there's any particular history for the name locale_update, but the other `Makefile` targets seem to be actions, e.g., generate_changelog, update_version, upload_xpi. So perhaps this should be `update_locale`?

File: docs/Releases.md [hunk 2]
```diff
@@ -1,12 +1,224 @@
...
+File a bug for merging the version of the add-on, e.g.
+[bug 1245912](https://bugzilla.mozilla.org/show_bug.cgi?id=1245912).
+
+```shell
+$ git checkout -b bug-<number>
+$ # Add/remove files as necessary
+$ git add ...
+$ git remove ...
+$ # For merges to mozilla-central, we are landing this with r=self for approved Firefox reviews.
+$ git commit -m "Bug <number> - Update Loop system add-on to version 0.3.0. r=<ircnick> for already reviewed code" -a
```
Comment (Mardak) [inline docs/Releases.md]: Might as well refer to `version <version>. r=<ircnick>...` to match.

File: docs/Releases.md [hunk 3]
```diff
@@ -1,12 +1,224 @@
...
+$ git-bz attach -n -e <number> HEAD
+```
+
+In the editor, uncomment `:me+`, add any extra notes, and let the patch upload.
+
+```shell
+$ # Now push to fx-team
+$ cd ../fx-team
+$ hg pull -u
+$ cd ../geck-dev
```
Comment (Mardak) [inline docs/Releases.md]: `gecko-dev` typo

File: docs/Releases.md [hunk 4]
```diff
@@ -1,12 +1,224 @@
...
+
+```shell
+$ make distclean
+$ make build
+$ make test
+```
+
+* Now run some manual tests:
+  * Use the standalone server: `make distserver`
+  * Run the add-on from the repo using `./bin/runfx.js`
```
Comment (Mardak) [inline docs/Releases.md]: `make runfx` to match the previous line?

File: Makefile (hunk 1 above)
Comment (Standard8) [inline Makefile:364]: Yeah, that sounds better.

**Pull request merged**